h/api.py

```diff
@@ -204,16 +202,19 @@ def delete(context, request):
 
 def get_user(request):
     """Create a User object for annotator-store"""
-    userid = request.authenticated_userid
+    userid = request.unauthenticated_userid
     if userid is not None:
-        consumer = auth.Consumer(request.client.client_id)
-        return auth.User(userid, consumer, False)
+        for principal in request.effective_principals:
+            if principal.startswith('consumer:'):
```

Comment:
I think this may be a transition issue, but while testing this locally I ran into the issue that `request.unauthenticated_userid` appears to be an integer, causing a `AttributeError` to be thrown by this line.

I'm not entirely sure where or when `userid` got set to an integer (presumably it should always be `user.username`) but we should probably handle this case.